common/namespace/cache.go

```diff
@@ -632,41 +583,67 @@ func (c *namespaceCache) triggerNamespaceChangeCallbackLocked(
 	}
 }
 
-func (c *namespaceCache) buildEntryFromRecord(
+func FromPersistentState(
+	thisCluster ClusterInfo,
 	record *persistence.GetNamespaceResponse,
 ) *CacheEntry {
-
-	// this is a shallow copy, but since the record is generated by persistence
-	// and only accessible here, it would be fine
-	newEntry := newCacheEntry(c.clusterMetadata)
-	newEntry.info = record.Namespace.Info
-	newEntry.config = record.Namespace.Config
-	newEntry.replicationConfig = record.Namespace.ReplicationConfig
-	newEntry.configVersion = record.Namespace.ConfigVersion
-	newEntry.failoverVersion = record.Namespace.FailoverVersion
-	newEntry.isGlobalNamespace = record.IsGlobalNamespace
-	newEntry.failoverNotificationVersion = record.Namespace.FailoverNotificationVersion
-	newEntry.notificationVersion = record.NotificationVersion
-	newEntry.initialized = true
-	return newEntry
+	return &CacheEntry{
+		thisCluster:                 thisCluster,
+		info:                        record.Namespace.Info,
+		config:                      record.Namespace.Config,
+		replicationConfig:           record.Namespace.ReplicationConfig,
+		configVersion:               record.Namespace.ConfigVersion,
+		failoverVersion:             record.Namespace.FailoverVersion,
+		isGlobalNamespace:           record.IsGlobalNamespace,
+		failoverNotificationVersion: record.Namespace.FailoverNotificationVersion,
+		notificationVersion:         record.NotificationVersion,
+		initialized:                 true,
+	}
+}
+
+type EntryMutation func(private, *persistence.GetNamespaceResponse)
```

Comment:
Why use `private` instead of making this type itself private?

Reply:
Updated to make this an interface with just an `apply` member function.